.github/workflows/build_cmake.yml

```diff
@@ -57,6 +57,11 @@ jobs:
     - name: Update Submodules
       run: git submodule update --init --recursive --depth=1
 
+    - name: Validate python style with black
+      run: |
+        pip install black
+        black source/codegen examples --check
```

Comment:
> Can this be run on local builds as well?

It's definitely possible to trigger this from cmake. It would be a challenge to list out all of the dependencies that require it to re-run. But, it runs fast enough, we could consider running it unconditionally. But I'd rather not add more logic to the cmake if we can help it.

> I don't see VSCode auto applying formatting even with black installed and configured. 

This should work and we really need to get it working. We need everyone that is working in this code to be set up with IDE enabled auto-formatting. I hope that any out-of-workflow changes that cause errors are relatively rare. This is also set up to run early in the workflow, so you'd get pretty quick feedback on error.

I can look at this some today and post a set of instructions. If you're still having trouble, I can help debug it.

We could also run `black --check`  in a pre-commit hook, that's pretty common. I think I'd prefer that to running as part of the build.

Reply:
I'm going to resolve this for now. We can discuss and setup workflow issues and make additions as needed.